File: trip-kit-booking/src/main/java/com/skedgo/tripkit/booking/QuickBookingServiceImpl.java
package com.skedgo.tripkit.booking;

import androidx.annotation.NonNull;

import java.util.List;

import io.reactivex.Observable;

@Deprecated
public class QuickBookingServiceImpl implements QuickBookingService {

  private final QuickBookingApi api;

  public QuickBookingServiceImpl(@NonNull QuickBookingApi api) {
    this.api = api;
  }

  @Override public Observable<List<QuickBooking>> fetchQuickBookingsAsync(@NonNull String url) {
    return api.fetchQuickBookingsAsync(url);
  }
}

File: trip-kit-booking/src/main/java/com/skedgo/tripkit/booking/QuickBookingServiceImpl.kt
package com.skedgo.tripkit.booking

import io.reactivex.Observable

@Deprecated("")
class QuickBookingServiceImpl(private val api: QuickBookingApi) : QuickBookingService {
    override fun fetchQuickBookingsAsync(url: String): Observable<List<QuickBooking>> {
        return api.fetchQuickBookingsAsync(url)
    }
}
